app/connectors/EventReportConnector.scala

```diff
@@ -136,21 +137,27 @@ class EventReportConnector @Inject()(
   def getEvent(pstr: String, startDate: String, version: String, eventType: EventType)
               (implicit headerCarrier: HeaderCarrier, ec: ExecutionContext): Future[JsValue] = {
 
-    val url = config.api1832Url.format(pstr)
+    val apiToCall = getApiTypeByEventType(eventType) match {
+      case Some(apiType) => apiType.toString
+      case None => ""
```

Comment:
Rather than creating an empty string here it might be nicer to just map the option and at end throw bad request exception at the end if None.

Reply:
For now I have added a thrown BadRequest exception on case None. This will only get reached during the development process for the various gets in this BE because after all of the cases are covered in getApiTypeByEventType the None case will be very hard to reach without coding in a nonsense eventType or if the list itself expands.